lib/sprinkle_dns/client.rb

```diff
@@ -89,32 +84,59 @@ def sprinkle!
         return [existing_hosted_zones, nil]
       end
 
+      hosted_zones = (existing_hosted_zones + missing_hosted_zones)
+
       unless @config.force?
-        changes = existing_hosted_zones.map{|h| SprinkleDNS::EntryPolicyService.new(h, @config)}.collect{|eps| eps.entries_to_change}.sum
-        puts
-        print "#{changes} changes to make. Continue? (y/N)"
+        changes = hosted_zones.map{|h| SprinkleDNS::EntryPolicyService.new(h, @config)}.collect{|eps| eps.entries_to_change}.sum
+
+        messages = []
+
+        messages << "#{missing_hosted_zones.size} hosted-zone(s) to create" if missing_hosted_zones.any?
+        messages << "#{changes} change(s) to make" if changes > 0
+        print messages.join(' and ').concat(". Continue? (y/N)")
+
         case gets.strip
         when 'y', 'Y'
           # continue
         else
           puts ".. exiting!"
-          return [existing_hosted_zones, nil]
+          return [hosted_zones, nil]
         end
       end
 
-      change_requests = @dns_provider.change_hosted_zones(existing_hosted_zones, @config)
       progress_printer = if @config.interactive_progress?
-        SprinkleDNS::CLI::InteractiveChangeRequestPrinter.new
+        SprinkleDNS::CLI::InteractiveChangeRequestPrinter
       else
-        SprinkleDNS::CLI::PropagatedChangeRequestPrinter.new
+        SprinkleDNS::CLI::PropagatedChangeRequestPrinter
       end
 
-      begin
-        @dns_provider.check_change_requests(change_requests)
-        progress_printer.draw(change_requests)
-      end until change_requests.all?{|cr| cr.in_sync}
+      # Create missing hosted zones
+      change_requests_hosted_zones = @dns_provider.create_hosted_zones(missing_hosted_zones)
```

Comment:
I'm not sure I like this flow, because if the error in this line happens, the user is not aware that we have already (maybe) send requests to AWS to make some changes. I'd prefer if we'd firstly printed "Creating hosted zones ...", and after that make a request.

Reply:
I understand that, but it is a bit hard to solve, I will have a think about it.